Sources/NIO/DatagramVectorReadManager.swift

```diff
@@ -100,7 +100,11 @@ struct DatagramVectorReadManager {
                 // TODO(cory): almost all of this except for the iovec could be done at allocation time. Maybe we should?
 
                 // First we set up the iovec and save it off.
+                #if os(Android)
+                self.ioVector[i] = IOVector(iov_base: bufferPointer.baseAddress! + (i * messageSize), iov_len: __kernel_size_t(messageSize))
```

Comment:
Rather than add this conditional code block, can we define an extension on the android iovec type that accepts the appropriate types in its initializer and does the conversion there? It'll clean up the code a lot.

Reply:
Will do.